contracts/RelayHub.sol

```diff
@@ -349,40 +367,45 @@ contract RelayHub is IRelayHub {
 
         // The recipient is no allowed to withdraw balance from RelayHub during a relayed transaction. We check pre and
         // post state to ensure this doesn't happen.
-        uint256 balanceBefore = balances[recipient];
+        atomicData.balanceBefore = balances[recipient];
 
         // First preRelayedCall is executed.
-        // It is the recipient's responsability to ensure, in acceptRelayedCall, that this call will not revert.
-        bytes32 preReturnValue;
         {
             // Note: we open a new block to avoid growing the stack too much.
             bytes memory data = abi.encodeWithSelector(
-                IRelayRecipient(recipient).preRelayedCall.selector,
-                relayAddr, from, encodedFunction, transactionFee
+                IRelayRecipient(recipient).preRelayedCall.selector, recipientContext
             );
 
+            // It is the recipient's responsibility to ensure, in acceptRelayedCall, that this call will not revert.
```

Comment:
What I tried to convey with that comment is that a recipient should check `pre`'s requirements in `accept` to make sure that it doesn't revert (because it will still be charged if it does). I've updated the wording to better reflect this.